eth/tracers/tracers.go

```diff
@@ -35,50 +42,31 @@ type Tracer interface {
 	Stop(err error)
 }
 
+type lookupFunc func(string, *Context) (Tracer, error)
+
 var (
-	nativeTracers map[string]func() Tracer = make(map[string]func() Tracer)
-	jsTracers                              = make(map[string]string)
+	lookups []lookupFunc
 )
 
-// RegisterNativeTracer makes native tracers which adhere
-// to the `Tracer` interface available to the rest of the codebase.
-// It is typically invoked in the `init()` function, e.g. see the `native/call.go`.
-func RegisterNativeTracer(name string, ctor func() Tracer) {
-	nativeTracers[name] = ctor
-}
-
-// New returns a new instance of a tracer,
-// 1. If 'code' is the name of a registered native tracer, then that tracer
-//   is instantiated and returned
-// 2. If 'code' is the name of a registered js-tracer, then that tracer is
-//   instantiated and returned
-// 3. Otherwise, the code is interpreted as the js code of a js-tracer, and
-//   is evaluated and returned.
-func New(code string, ctx *Context) (Tracer, error) {
-	// Resolve native tracer
-	if fn, ok := nativeTracers[code]; ok {
-		return fn(), nil
+// RegisterLookup registers a method as a lookup for tracers, meaning that
+// users can invoke a named tracer through that lookup. If 'wildcard' is true,
+// then the lookup will be placed last. This is typically meant for interpreted
+// engines (js) which can evaluate dynamic user-supplied code.
+func RegisterLookup(wildcard bool, lookup lookupFunc) {
+	if wildcard {
```

Comment:
Just an observation that this won't work if there are e.g. 2 JS engines (one will starve). Not that we'll ever have 2. But I guess generally I never liked that we're using one parameter to either select from given tracers or to evaluate a JS script. I'd have preferred sth like `{jscode: "{...}"}` then we could have a separate function for initializing a tracer with dynamic code.

Reply:
Yes, we can only have one 'wildcard-matcher'. In hindsight, yes, it would have been nicer with a marker for it, e.g `//javascript\n`